### PR TITLE
fix: count month days for teams challenge days left

### PR DESCRIPTION
**Pull request by christian14b**
`IntervalToDuration` method returns an interval object with ({years, months, days}) we were just getting the days between two dates causing an issue if the difference between them is 1 month at least.

Note: I'm assuming challenge's days are less than 1 year for this fix

Given these dates:
Start: today
End: `2024-07-17T15:37:46.711Z`

## Issue
image

## Fix:
image

### CODE REVIEW

File: src/plugins/team-goal-mixin.js [hunk 1]
```diff
@@ -48,10 +48,17 @@ export default {
...
-			return intervalToDuration({
+			const interval = intervalToDuration({
 				start,
 				end,
-			}).days;
+			});
+
+			if (interval?.months) {
+				const daysInMonth = getDaysInMonth(start);
+				return interval.days + (interval.months * daysInMonth);
```
Comment (roger-in-kiva) [inline src/plugins/team-goal-mixin.js:58]: Not sure if we should cover this right now, but what happens if we have more than 1 month?

File: src/plugins/team-goal-mixin.js [hunk 2]
```diff
@@ -48,10 +48,17 @@ export default {
 		daysLeft() {
 			const start = this.goal?.startDate ? new Date(this.goal?.startDate) : new Date();
 			const end = this.goal?.endDate ? new Date(this.goal?.endDate) : new Date();
-			return intervalToDuration({
+			const interval = intervalToDuration({
```
Comment (mcstover) [inline src/plugins/team-goal-mixin.js:51]: I think we may want to use one of the "days" based methods instead. Maybe [differenceInDays](https://date-fns.org/v3.6.0/docs/differenceInDays)

Comment (mcstover) [reply src/plugins/team-goal-mixin.js:51]: That said, there is still opportunity with both your current method and the new one to have an offset in hours due to timezones when using dates from the goal service which are static...

**Comment by emuvente** [comment]
I think formatDistanceToNowStrict will do what you want here, no matter the dates. https://date-fns.org/v3.6.0/docs/formatDistanceToNowStrict

**Comment by mcstover** [comment]
This exploration of date-fns methods will help use refine our clientside date handling in the impact-dashboard too ;)

**Review by mcstover — APPROVED** [review]
Neat!

**Pull request merged**